drivers/hmis/app/models/hmis/form/definition.rb

```diff
@@ -508,26 +511,37 @@ def external_form_object_key=(value)
     super(value.blank? ? nil : value.strip)
   end
 
+  # Walk definition items as OpenStructs
```

Comment:
We have both walk_definition_nodes and walk_definition_items. Could we remove walk_definition_nodes at this point? Looks like we'd just need to adjust `introspect_custom_data_element_definitions`

Reply:
The open struct options is nice for read-only traversal, which I feel like might be useful elsewhere. I combined them into one outer method https://github.com/greenriver/hmis-warehouse/pull/4480/commits/89bd6ae152de698ac33866e499fd5ea18b41619c how's that?